File: src/main/java/frc/robot/subsystems/drivebase/GyroIOSim.java
package frc.robot.subsystems.drivebase;


import edu.wpi.first.math.geometry.Rotation2d;

public class GyroIOSim extends GyroIO {
    private Rotation2d rotationOffset = new Rotation2d();

    @Override
    public void updateInputs(GyroIOInputs inputs) {
        inputs.yaw = DrivebaseIOSim.getHeading().minus(rotationOffset);
    }

    @Override
    public void setYaw(double degrees) {
        rotationOffset = DrivebaseIOSim.getHeading().plus(Rotation2d.fromDegrees(degrees));
    }
}

File: src/main/java/frc/robot/subsystems/drivebase/GyroIOSim.kt
package frc.robot.subsystems.drivebase

import edu.wpi.first.math.geometry.Rotation2d


class GyroIOSim : GyroIO() {
    private var rotationOffset = Rotation2d()

    override fun updateInputs(inputs: GyroIOInputs) {
        inputs.yaw = DrivebaseIOSim.heading.minus(rotationOffset)
    }

    override fun setYaw(yaw: Rotation2d) {
        rotationOffset = DrivebaseIOSim.heading.minus(yaw)
    }
}
